File: Medium/2026-02-26-1404-Number-of-Steps-to-Reduce-a-Number-in-Binary-Representation-to-One/solution.py
class Solution:
    def numSteps(self, s: str) -> int:
        steps = 0
        while s != "1":
            if s[-1] == '0':
                s = s[:-1]  # Remove the last character (divide by 2)
            else:
                s = bin(int(s, 2) + 1)[2:]  # Add 1 and convert back to binary
            steps += 1
        return steps

File: Medium/2026-02-26-1404-Number-of-Steps-to-Reduce-a-Number-in-Binary-Representation-to-One/solution.java
class Solution {
    public int numSteps(String s) {
        int steps = 0;
        int n = s.length();
        
        for (int i = n - 1; i > 0; i--) {
            if (s.charAt(i) == '1') {
                steps += 2; // Add 1 and then divide by 2
            } else {
                steps += 1; // Just divide by 2
            }
        }
        
        return steps;
    }
}